cloud-aws-common/src/test/java/com/sequenceiq/cloudbreak/cloud/aws/common/consumption/AwsEFSConsumptionCalculatorTest.java

```diff
@@ -0,0 +1,158 @@
+                .withFileSystems(List.of());
+        when(awsEfsCommonService.getEfsSize(null, REGION_NAME, startTime, endTime, BUCKET_NAME)).thenReturn(statisticsResult);
+
+        CloudConnectorException ex = assertThrows(CloudConnectorException.class, () -> underTest.calculate(request));
+
+        verify(awsEfsCommonService).getEfsSize(null, REGION_NAME, startTime, endTime, BUCKET_NAME);
+        assertEquals(String.format("No Efs were returned by efs id %s and timeframe from %s to %s",
+                BUCKET_NAME, startTime, endTime), ex.getMessage());
+    }
+
+    @Test
+    public void getObjectStorageSizeOneDatapoint() {
+        Date startTime = Date.from(Instant.now().minus(42, ChronoUnit.MINUTES));
+        Date endTime = Date.from(Instant.now());
+        Region region = Region.region(REGION_NAME);
+        StorageSizeRequest request = StorageSizeRequest.builder()
+                .withObjectStoragePath(BUCKET_NAME)
+                .withStartTime(startTime)
+                .withEndTime(endTime)
+                .withRegion(region)
+                .build();
+
+        FileSystemDescription description = new FileSystemDescription()
+                .withCreationTime(Date.from(Instant.now()))
+                .withSizeInBytes(new FileSystemSize().withValue(42L));
+        DescribeFileSystemsResult statisticsResult = new DescribeFileSystemsResult()
+                .withFileSystems(List.of(description));
+        when(awsEfsCommonService.getEfsSize(null, REGION_NAME, startTime, endTime, BUCKET_NAME)).thenReturn(statisticsResult);
+
+        StorageSizeResponse result = underTest.calculate(request);
+
+        verify(awsEfsCommonService).getEfsSize(null, REGION_NAME, startTime, endTime, BUCKET_NAME);
+        assertEquals(42.0, result.getStorageInBytes(), DOUBLE_ASSERT_EPSILON);
+    }
+
+    @Test
+    public void getObjectStorageSizeLatestDatapointIsUsed() {
+        Date startTime = Date.from(Instant.now().minus(42, ChronoUnit.MINUTES));
+        Date endTime = Date.from(Instant.now());
+        Region region = Region.region(REGION_NAME);
+        StorageSizeRequest request = StorageSizeRequest.builder()
+                .withObjectStoragePath(BUCKET_NAME)
+                .withStartTime(startTime)
+                .withEndTime(endTime)
+                .withRegion(region)
+                .build();
+
+        FileSystemDescription latestDatapoint = new FileSystemDescription()
+                .withCreationTime(Date.from(Instant.now()))
+                .withSizeInBytes(new FileSystemSize().withValue(42L));
+        FileSystemDescription earlierDatapoint = new FileSystemDescription()
+                .withCreationTime(Date.from(Instant.now()))
+                .withSizeInBytes(new FileSystemSize().withValue(12L));
+        FileSystemDescription earliestDatapoint = new FileSystemDescription()
+                .withCreationTime(Date.from(Instant.now()))
+                .withSizeInBytes(new FileSystemSize().withValue(22L));
```

Comment:
The description with the value 42L, would in fact be the earliest datapoint in this case

Reply:
in case of EFS this test is not really make sense